lib/Sema/TypeCheckStmt.cpp

```diff
@@ -2599,11 +2611,6 @@ bool TypeCheckASTNodeAtLocRequest::evaluate(
         // thus the transform couldn't be applied. Perform code completion
         // pretending there was no result builder to recover.
       }
-    } else if (func->hasSingleExpressionBody() &&
-                func->getResultInterfaceType()->isVoid()) {
-       // The function returns void.  We don't need an explicit return, no matter
-       // what the type of the expression is.  Take the inserted return back out.
-      func->getBody()->setLastElement(func->getSingleExpressionBody());
```

Comment:
Was this simply no longer needed or was it rendered unnecessary by one of your changes?

Reply:
This is now done by PreCheckFunctionBodyRequest

Reply:
Oh, I missed that. Sorry